common/sealable-trie/src/bits/concat.rs

```diff
@@ -0,0 +1,455 @@
+use alloc::vec::Vec;
+
+use lib::u3::U3;
+
+use super::{Owned, Slice};
+
+
+/// Trying to concatenate slices which result in slice whose size is too large.
```

Comment:
why do we have to implement this? no existing library can give us these features?

Reply:
Honestly we probably can use [bitvec](https://docs.rs/bitvec/latest/bitvec/index.html).  Rewriting the code is just something I haven’t gotten to doing.  The bits module grew organically to fit the needs of the trie where each change was relatively simple.  Because of that each time just adding another small feature seemed like less work than rewriting everything to use bitvec.  But it’s definitely something I need to look into at some point and definitely before prod release.